server/models/Checklist.js

```diff
@@ -0,0 +1,116 @@
+const mongoose = require("mongoose");
+const BasePlugin = require("./BasePlugin");
+
+const checklistItemSchema = new mongoose.Schema({
+  checklistItem: {
+    type: String,
+  },
+  isChecked: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const ChecklistPluginSchema = BasePlugin.discriminator(
+  "ChecklistPlugin",
+  new mongoose.Schema(
+    {
+      checklist: [checklistItemSchema],
+    },
+    { timestamps: true }
+  )
+);
+
+ChecklistPluginSchema.methods.get = async ({ params }) => {
+  const { checklistItemId } = params;
+  if (!checklistItemId) {
+    throw new Error("checklistItemId is not defined");
+  }
+  const data = await this.model("ChecklistPlugin").findById(checklistItemId);
+  if (data) {
+    return { status: 200, data };
+  }
+  throw new Error("something went wrong");
+};
+
+ChecklistPluginSchema.methods.create = async ({ body }) => {
+  const { checklistItem } = body;
+  if (!checklistItem) {
+    throw new Error("checklistItem is not defined");
+  }
+
+  const data = await mongoose.model("ChecklistPlugin").create({
+    checklistItem,
+    isChecked: false,
+  });
+  if (data) {
+    return {
+      status: 200,
+      data,
+    };
+  }
+  throw new Error("something went wrong");
+};
+
+ChecklistPluginSchema.methods.update = async ({ body }) => {
+  const { checklistItem, checklistItemId, isChecked } = body;
+  if (!checklistItemId || !isChecked || !checklistItem) {
+    throw new Error("required params is not defined");
+  }
```

Comment:
We should be responding with an object

`{ response: XXX, message: "" }`

Please address where applicable